xbmc/filesystem/Directory.cpp

```diff
@@ -183,17 +183,15 @@ bool CDirectory::GetDirectory(const CURL& url,
         g_directoryCache.ClearDirectory(realURL);
 
       pDirectory->SetFlags(hints.flags);
+      items.SetURL(url);
 
       bool result = false;
       CURL authUrl = realURL;
 
       while (!result)
       {
         // don't change auth if it's set explicitly
-        if (CPasswordManager::GetInstance().IsURLSupported(authUrl) && authUrl.GetUserName().empty())
-          CPasswordManager::GetInstance().AuthenticateURL(authUrl);
-
-        items.SetURL(url);
+        authUrl = URIUtils::AddCredentials(authUrl);
```

Comment:
If you want to save on copies then you should use `std::move`:
```suggestion
        authUrl = URIUtils::AddCredentials(std::move(authUrl));
```
There are a couple more locations below  where it would make sense.

Reply:
@neo1973 Done.

Do you have any thoughts on trailing moved from types. (An example would be better)

```
void foo() {
  std::vector<int> something;
  use_something( something ); // fine
  move_something( something ); // fine
  // ... 20+ more lines in the function
}
```

I have ensured that `something` is not used after move.... But another developer could come along at a later point and modify one of the 20+ lines (using `something`) without realising the object has been moved from. 

I know it is up to us developers not to make this kind of mistake; but it is also up to us developers to also leave code minimising any future risk.

Interested in hearing your thoughts.